src/messaging/MessageEnvelope.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { DecryptCallback, DidResolveKey, DidResourceUri, EncryptCallback } from '@kiltprotocol/types'
+import * as Did from '@kiltprotocol/did'
+import * as MessageError from './Error'
+import { hexToU8a, stringToU8a, u8aToHex, u8aToString } from '@polkadot/util'
+
+import type { IEncryptedMessage, IEncryptedMessageContents, IMessage } from '../types'
+
+/**
+ * Checks if the message object is well-formed.
+ *
+ * @param message The message object.
+ */
+export function verifyMessageEnvelope(message: IMessage): void {
```

Comment:
should this be hidden from the public interface and simply called on `encrypt` and `decrypt`?

Reply:
tada: [e06c1ec](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/e06c1ecee069939a21d1324cb5dbc0b44fa48399)